source/fab/tools/tool.py

```diff
@@ -28,17 +27,25 @@ class Tool:
     the name of the executable, and provides a `run` method.
 
     :param name: name of the tool.
-    :param exec_name: name of the executable to start.
+    :param exec_name: name or full path of the executable to start.
     :param category: the Category to which this tool belongs.
+    :param availability_option: a command line option for the tool to test
+        if the tool is available on the current system. Defaults to
+        `--version`.
     '''
 
-    def __init__(self, name: str, exec_name: str,
-                 category: Categories = Categories.MISC):
+    def __init__(self, name: str, exec_name: Union[str, Path],
+                 category: Category = Category.MISC,
+                 availablility_option: Optional[str] = None):
         self._logger = logging.getLogger(__name__)
         self._name = name
-        self._exec_name = exec_name
+        self._exec_name = str(exec_name)
```

Comment:
At the moment this is fine since the executable filename is only ever used to compile the argument list, which is a list of strings. If we ever need to modify the path or perform tests on it we may prefer to keep it as a `Path`. e.g. `self._exec_name.is_executable()`.

Reply:
OK. It would be easy to push a path through if needed.